machine/machine.go

```diff
@@ -101,6 +104,57 @@ func GetMachineMemoryCapacity() (uint64, error) {
 	return memoryCapacity, err
 }
 
+// GetMachineMemoryByType returns information about memory capcity and number of DIMMs.
+// Information is retrieved from sysfs edac per-DIMM API (/sys/devices/system/edac/mc/)
+// introduced in kernel 3.6. Documentation can be found at
+// https://www.kernel.org/doc/Documentation/admin-guide/ras.rst.
+func GetMachineMemoryByType(edacPath string) (map[string]*info.MemoryInfo, error) {
+	memory := map[string]*info.MemoryInfo{}
+	names, err := ioutil.ReadDir(edacPath)
+	if err != nil {
+		return memory, err
+	}
+	isController := regexp.MustCompile("mc[0-9]+")
+	isDimm := regexp.MustCompile("dimm[0-9]+")
+	for _, controllerDir := range names {
+		controller := controllerDir.Name()
+		if isController.MatchString(controller) {
+			dimms, err := ioutil.ReadDir(path.Join(edacPath, controllerDir.Name()))
+			if err != nil {
+				return map[string]*info.MemoryInfo{}, err
```

Comment:
inconsistency previously `memory` is returned instead of  `map[string]*info.MemoryInfo{}`

Reply:
It was intentional - I wanted to avoid sending incomplete memory information.